### PR TITLE
[exporter/datadog] Add 'usePreview' option when the hostname comes from attributes

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Add a (currently internal) `usePreview` option to use the improved hostname mapping when getting the hostname from attributes. Eventually whether to use the old/new logic will be controlled by a feature flag, but right now it is unused.

**Link to tracking Issue:** #10424

**Testing:** Improved existing unit tests to avoid regressions

### CODE REVIEW

File: exporter/datadogexporter/internal/model/attributes/hostname_test.go [hunk 1]
```diff
@@ -34,91 +35,125 @@ const (
...
+		{
+			name: "azure",
+			attrs: testutils.NewAttributeMap(map[string]string{
+				conventions.AttributeCloudProvider: conventions.AttributeCloudProviderAzure,
+				conventions.AttributeHostID:        testHostID,
+				conventions.AttributeHostName:      testHostName,
+			}),
+			ok:       true,
+			hostname: testHostName,
+		},
```
Comment (KSerrania) [inline exporter/datadogexporter/internal/model/attributes/hostname_test.go:127]: We have EC2 tests with preview above, but we don't have them for GCP and Azure. Is that expected?

Comment (mx-psi) [reply exporter/datadogexporter/internal/model/attributes/hostname_test.go]: The tests here duplicate the tests on the gcp/azure/ec2 folders, but it doesn't hurt to have them so I added them too.

File: exporter/datadogexporter/internal/model/attributes/hostname_test.go [hunk 2]
```diff
@@ -18,107 +18,167 @@ import (
...
+			name: "GCP, preview",
+			attrs: testutils.NewAttributeMap(map[string]string{
+				conventions.AttributeCloudProvider:  conventions.AttributeCloudProviderGCP,
+				conventions.AttributeHostID:         testHostID,
+				conventions.AttributeHostName:       testGCPHostname,
+				conventions.AttributeCloudAccountID: testCloudAccount,
+			}),
+			usePreview: true,
+			ok:         true,
+			hostname:   testHostName + "." + testCloudAccount,
```
Comment (KSerrania) [inline exporter/datadogexporter/internal/model/attributes/hostname_test.go:130]: Nit: I'd move this to a variable named `testGCPIntegrationHostname` for better legibility.

**Review by KSerrania — APPROVED** [review]
LGTM

**Review by codeboten — APPROVED** [review]

**Pull request merged**